### DISCUSSION TITLE
Have to refresh 3+ times to get search result

### DISCUSSION THREAD

**Discussion by cacharle** [searx/searx#3457] [General]
I have a private instance of Searx but when I search anything, I generally have to refresh the page 3+ times to get the search results.

The first time, I have a red message box with `Engines cannot retrieve results:` with all my engines

Then I have a blue box with no results for a few refresh.

Then I have the results for all my engines, and when I try with another query 10s later, it works right away.

Any idea why my instance needs to "warm up" like that?

**Comment by cacharle** [comment]
![Screenshot 2023-02-02 at 11 02 55](https://user-images.githubusercontent.com/25375061/216294402-29d3daa6-f548-47ef-bccb-e4752a9b8369.png)

**Comment by unixfox** [comment]
https://github.com/searx/searx/issues/3042

**Reply by cacharle** [reply]
I'm not sure increasing the timeout would make a difference since I get the red error instantaneously.

From what I understand from this issue, the reason it works after the first time is because searx keep the connection active.

**Reply by unixfox** [reply]
You should try this fork of searx it works much better: https://github.com/searxng/searxng

**Reply by cacharle** [reply]
:/ indeed, I was afraid I would have to switch to searxng..

I thought I didn't need to because it says in the README that it's made for public instances (mine is private)

**Reply by unixfox** [reply]
It's made for everyone, it's a lie that it's only for public instances.
You should read the actual diff from the README of searxng: https://github.com/searxng/searxng#differences-to-searx

**Reply by cacharle** [reply]
Oh, sorry I read that somewhere else apparently.

**Reply by cacharle** [reply]
Ok I thought using searxng would take a lot of time to setup but it took me 10min and work flawlessly so far :smile:, ty for the recommandation

**Comment by cacharle** [comment]
![image](https://user-images.githubusercontent.com/25375061/216597246-d866f129-79e4-43f1-aef1-90618c79a3b0.png)

**Comment by cacharle** [comment]
![image](https://user-images.githubusercontent.com/25375061/217009365-146d46e3-47c5-4a2e-8e18-217d182899d7.png)

I don't understand, I still get the blue message even if I have 3000+ results.
